src/main/kotlin/org/rust/lang/core/types/infer/Declarations.kt

```diff
@@ -80,3 +82,17 @@ fun inferTypeReferenceType(ref: RsTypeReference): Ty {
         else -> TyUnknown
     }
 }
+
+internal fun resolveLifetime(element: RsElement): Re {
```

Comment:
This will probably be more readable if rewritten with `when`, like this:
```kotlin
    val lifetime = when (element) {
        is RsRefLikeType -> element.lifetime
        is RsSelfParameter -> element.lifetime
        else -> null
    } ?: return ReUnknown

    val lifetimeName = lifetime.referenceName

    when (lifetimeName) {
        "'_" -> return ReUnderscore
        "'static" -> return ReStatic
    }

    val resolved = lifetime.reference.resolve()
    return when (resolved) {
        is RsLifetimeParameter -> ReEarlyBound.named(resolved)
        else -> ReUnknown
    }
```

Reply:
Let's use `if` expr for binary condition instead of `when`
```kotlin
val resolved = lifetime.reference.resolve()
return if (resolved is RsLifetimeParameter) ReEarlyBound.named(resolved) else ReUnknown
```

Reply:
@Undin I dislike this syntax very much :D
IMO, it is badly readable
Maybe spliting branches on separate lines will be better?

Reply:
@fedochet I agree that one-line `if` expr is not readable if child expressions are long and in such cases I'd prefer to split it into different lines
But in this case it looks like good enough